src/toggle-switch/toggle-switch.scss

```diff
@@ -94,47 +110,58 @@
           border-color: t(iui-color-background-primary);
         }
 
-        > .iui-icon {
-          @include themed {
-            opacity: t(iui-opacity-1);
-          }
-        }
-
-        > .iui-handle {
+        &::after {
+          left: $iui-m + $iui-xs + 1px;
           @include themed {
             background-color: t(iui-color-foreground-accessory);
             opacity: t(iui-opacity-2);
           }
-          right: $iui-xxs;
         }
-      }
 
-      &:hover:not(:disabled) {
-        ~ .iui-toggle {
+        > .iui-icon {
           @include themed {
-            border-color: t(iui-color-background-primary);
+            opacity: t(iui-opacity-1);
           }
+        }
+      }
 
-          > .iui-handle {
-            @include themed {
-              opacity: t(iui-opacity-1);
-            }
-          }
+      &:hover:enabled ~ .iui-toggle {
+        @include themed {
+          border-color: t(iui-color-background-primary);
         }
       }
     }
 
+    // Focused or hovered toggle switch
+    &:hover:enabled ~ .iui-toggle::after,
+    &:focus ~ .iui-toggle::after {
+      @include themed {
+        opacity: t(iui-opacity-1);
+      }
+    }
+
+    // Focused toggle switch
+    &:focus ~ .iui-toggle {
+      outline-offset: 1px;
+      @include themed {
+        outline: 1px solid t(iui-color-foreground-primary);
+      }
+    }
+
+    &:focus:not(:focus-visible) ~ .iui-toggle {
+      outline: none;
+    }
+
     // Disabled toggle switch
     &:disabled {
       ~ .iui-toggle {
         cursor: not-allowed;
-
         @include themed {
           background-color: t(iui-color-background-disabled);
           border-color: t(iui-color-background-disabled);
         }
 
-        > .iui-handle {
+        &::after {
           @include themed {
             background-color: t(iui-color-foreground-body);
             opacity: t(iui-opacity-5);
```

Comment:
Was not allowed to comment, but on line `180` cursor not-allowed for iui-label might not be needed, as its inherited.

Reply:
Seems you are correct, removed.